persist/sqldb/workflow_archive.go

```diff
@@ -163,16 +163,37 @@ func (r *workflowArchive) ListWorkflows(options sutils.ListOptions) (wfv1.Workfl
 		return nil, err
 	}
 
-	selector := r.session.SQL().
-		Select(selectQuery).
+	subSelector := r.session.SQL().
+		Select(db.Raw("uid")).
 		From(archiveTableName).
 		Where(r.clusterManagedNamespaceAndInstanceID())
 
-	selector, err = BuildArchivedWorkflowSelector(selector, archiveTableName, archiveLabelsTableName, r.dbType, options, false)
+	subSelector, err = BuildArchivedWorkflowSelector(subSelector, archiveTableName, archiveLabelsTableName, r.dbType, options, false)
 	if err != nil {
 		return nil, err
 	}
 
+	if r.dbType == MySQL {
+		// workaround for mysql 42000 error (Unsupported subquery syntax):
+		//
+		//     Error 1235 (42000): This version of MySQL doesn't yet support 'LIMIT \u0026 IN/ALL/ANY/SOME subquery'
+		//
+		// more context:
+		// * https://dev.mysql.com/doc/refman/8.0/en/subquery-errors.html
+		// * https://dev.to/gkoniaris/limit-mysql-subquery-results-inside-a-where-in-clause-using-laravel-s-eloquent-orm-26en
+		subSelector = r.session.SQL().Select(db.Raw("*")).From(subSelector).As("x")
+	}
+
+	// why using a subquery: the json unmarshaling process of coalesce
+	// (workflow) is always triggered for every records in the filtering query
+	// in postgres (and maybe mysql as well?). by first filtering out uid first
+	// without trying to unmarshal the workflow, it kinda teach the database to
+	// delay the json parsing operation and thus speed up the query (could be
+	// up to 257 times faster for some deployments).
```

Comment:
```suggestion
	// why a subquery? the json unmarshal triggers for every row in the filter query.
	// by filtering on uid first, we delay json parsing until a single row, speeding up the query
	// (e.g. up to 257 times faster for some deployments).
```

can simplify this a little